### DISCUSSION TITLE
Is there a "print out controller state changes" mode?

### DISCUSSION THREAD

**Discussion by jaysonlarose** [darthcloud/BlueRetro#255] [Ideas]
Hi, Jacques!

I had a hard time remapping my controller, and I was wondering if there was a way to put BlueRetro into a state where it dumps out the controller state to the USB-Serial line?  That way I can press a button or move an axis and learn what the BlueRetro name for that control is.

This would also allow some other functionality I'm interested in, such as controller visualization and triggering actions on the host PC with unmapped buttons.

**Comment by darthcloud** [comment]
You can use the "BlueRetro_pad_test.bin" firmware to display most buttons state to the serial console.

But this is only indented for testing BT controller and do not route the state to the wired side.

Also make sure to to config your terminal properly to make it readable:
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10
"Implicit CR in every LF" is the important part.

Visualization is something I had in mind, but for that I would likely just send the raw generic data struct which is much less data than the current formatted debug mode.

This current test mode may add significant lag and I wouldn't play with that on.
